Convert2RHEL role (#1286)
theforeman/foreman-ansible-modules: roles/convert2rhel/README.md

theforeman.foreman.convert2rhel

> ===========================

Reviewer: ```suggestion
===============================
```

This role creates a basic configuration for everything needed to register and convert CentOS clients to Red hat Enterprise Linux.

> First is upload of manifest & sync of RHEL 7 & RHEL 8 repositories, for more detail see [content_rhel Role](https://github.com/theforeman/foreman-ansible-modules/blob/develop/roles/content_rhel/README.md).

Reviewer: I can't exactly parse that sentence. Did you want to say "First it uploads a manifest and syncs RHEL, …"?

> The role creates then Convert2RHEL products & repositories (with synchronization), activation keys and host groups for each OS.

Reviewer: ```suggestion
The role then creates Convert2RHEL products & repositories (and synchronizes them), activation keys and host groups for each OS.
```

If simple content access is disabled, subscriptions and repositories for RHEL activation keys must be added manually.

Role Variables
--------------

This role supports the [Common Role Variables](https://github.com/theforeman/foreman-ansible-modules/blob/develop/README.md#common-role-variables) and [Content RHEL variables](https://github.com/theforeman/foreman-ansible-modules/blob/develop/roles/content_rhel/README.md)

> - `convert2rhel_lifecycle_env`: Lifecycle environment for activation keys, default: Library.
> - `convert2rhel_content_view`: Content view for activation keys, default: Default Organization View.

Reviewer: ```suggestion
- `foreman_convert2rhel_lifecycle_env`: Lifecycle environment for activation keys, default: Library.
- `foreman_convert2rhel_content_view`: Content view for activation keys, default: Default Organization View.
```

in `roles/convert2rhel/defaults/main.yml` there are more variables, should they also be documented?

theforeman.foreman.convert2rhel
===========================

> This role creates a basic configuration for everything needed to register and convert `CentOS` clients to `Red hat Enterprise Linux`.

Reviewer: I don't think we should format CentOS an RHEL as code here?

First is upload of manifest & sync of `RHEL 7` & `RHEL 8` repositories, for more detail see [content_rhel Role](https://github.com/theforeman/foreman-ansible-modules/blob/develop/roles/content_rhel/README.md).

The role creates then Convert2RHEL products & repositories (with synchronization), activation keys and host groups for each OS.

Role Variables
--------------

This role supports the [Common Role Variables](https://github.com/theforeman/foreman-ansible-modules/blob/develop/README.md#common-role-variables).

> - `foreman_c2r_simple_content_access`: TODO: Simple content access enabled.

Reviewer: I'd prefer if the prefix would be `foreman_convert2rhel`, not `…_c2r`

- `foreman_c2r_manifest_path`: Path to manifest, default: `~/manifest.zip`.
- `foreman_c2r_enable_rhel7`: Enable & sync RHEL 7 repositories, default: `true`.
- `foreman_c2r_enable_rhel8`: Enable & sync RHEL 8 repositories, default: `true`.

> - `foreman_c2r_rhel8_releasever`: RHEL 8 repository version, required if `foreman_c2r_enable_rhel8` is `true`.

Reviewer: is there a reason to have these "redefined" instead of using the defaults of `content_rhel` and let the user override the `conent_rhel` values directly?
